python-spec/src/somacore/ephemeral/collections.py

```diff
@@ -154,6 +162,32 @@ def transformations(self) -> MutableMapping[str, coordinates.CoordinateTransform
         raise NotImplementedError()
 
 
+class Image2D(  # type: ignore[misc]   # __eq__ false positive
+    BaseCollection[base.SOMAObject], images.Image2D[data.DenseNDArray, base.SOMAObject]
+):
+    """An in-memory Collection with Image2D semantics."""
+
+    __slots__ = ()
+
+    @property
+    def level_count(self) -> int:
+        raise NotImplementedError()
+
+    def level_properties(self, level: int) -> images.Image2D.LevelProperties:
+        raise NotImplementedError()
+
+    def read_level(
+        self,
+        level: int,
+        coords: options.DenseNDCoords = (),
+        *,
+        transform: Optional[coordinates.CoordinateTransform] = None,
+        result_order: options.ResultOrderStr = options.ResultOrder.AUTO,
+        platform_config: Optional[options.PlatformConfig] = None,
+    ) -> pa.Tensor:
+        raise NotImplementedError()
```

Comment:
This function will be modified after I create a return type for spatial reads. The new return type will include coordinate space information and the ability to lazily apply coordinate transforms.